app/controllers/quiz_responses_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class QuizResponsesController < ApplicationController
+  before_action :authenticate_admin, only: %i[create destroy update]
+  before_action :find_quiz_response, only: %i[destroy links show steps update]
+
+  def show
+    render json: @quiz_response
+  end
+
+  def create
+    @quiz_response = QuizResponse.new(quiz_question_id: params[:quiz_question_id])
+
+    begin
+      @quiz_response.save!
+    rescue ActiveRecord::RecordInvalid => e
+      render status: 400, json: { error: e.message } and return
+    end
+    render status: 201, json: @quiz_response
+  end
```

Comment:
Since this is basically an if / else, it might be clearer to understand in that format:

```

if @quiz_response.save
  render status: 201, json: @quiz_response
else
  render status: 400, json: { error: @quiz_response.errors.full_messages }
end
    
```

Reply:
(I see this below in other places, too. If that's the structure you prefer for handling these sorts of actions, all good)

Reply:
I like that! Thank you :)